### PR TITLE
Improve documentation in Thentos.Transaction.TH

### LINKED ISSUE #189: Document Transaction.TH
Since #184, calling `makeThentosAcidicPhase{1,2}` is quite involved.  I would like to have haddock documentation on the module `Thentos.Transaction.TH` that links to the concept paper, explains how transactions are added to the core or how this module is to be used by library users, exposes the -ddump-splices material (currently in an internal comment) to the haddock output, ...

### PR DESCRIPTION
**Pull request by fhartwig**
Fixes #189

### CODE REVIEW

File: thentos-core/src/Thentos/Transaction/TH.hs [hunk 1]
```diff
@@ -26,41 +29,80 @@ import Thentos.Types (ThentosError, Extends)
...
-    acidEvents = dbEvents
--}
+-- | This function is responsible for making a user-defined db type and a set
+-- of transactions (either user-defined or provided by Thentos) work with
+-- acid-state. For a detailed description of the problem and how we solve it,
+-- read the documentation at
+-- <https://github.com/liqd/thentos/blob/master/docs/concepts/AcidPoly.lhs>.
+--
+-- The first argument is the db type which we want to make acidic. The second
+-- argument is a list if all the new (i.e. not inherited) transactions on that
```
Comment (jkarni) [inline thentos-core/src/Thentos/Transaction/TH.hs]: "list if" --> "list of"?

File: thentos-core/src/Thentos/Transaction/TH.hs [hunk 2]
```diff
@@ -260,9 +302,9 @@ dropPrefix (nameBase -> s)
 countArgs :: Type -> Int
 countArgs = length . fst . extractArgs
 
--- | Convert e.g. @a -> b -> ThentosUpdate Foo@ to
--- @a -> b -> Update DB (Either (ThentosError DB) Foo)@ and check whether it
--- is an Update or a Query.
+-- | Convert e.g. @(db `Extends` DB) a -> b -> ThentosUpdate db Foo@ to
```
Comment (jkarni) [inline thentos-core/src/Thentos/Transaction/TH.hs]: Missing a `=>`

File: thentos-core/src/Thentos/Transaction/TH.hs [hunk 3]
```diff
@@ -26,41 +29,80 @@ import Thentos.Types (ThentosError, Extends)
...
 
+-- | This function takes a database type and a list of user-defined transactions
+-- that may look like this:
+--
+-- > trans_foo :: (db `Extends` MyDB) => Bool -> ThentosQuery db Int
+--
+-- (where MyDB is the passed-in db type).
+-- and generates transactions of the form:
+-- 
+-- > (db `Extends` MyDB) =>
```
Comment (jkarni) [inline thentos-core/src/Thentos/Transaction/TH.hs]: Maybe `foo :: ...` so it's analogous to the one above, and it's more obvious how the functions are renamed?

**Comment by fhartwig** [comment]
@jkarni Good points, thank you. Fixed.

**Pull request merged**